### PR TITLE
WIP - Added loaders.py (DataLoader)

### PR DESCRIPTION
**Pull request by kulothunganug**
### Description of Changes
A modified version of dataloader from [akivymd](https://github.com/quitegreensky/akivymd)

### Usage

```py
from kivy.lang import Builder
from kivymd.app import MDApp

KV = """

Screen:
	MDImageLoader:
		id: img
		size_hint: None,None
		size: dp(125), dp(125)
		pos_hint: {"center_x": .5, "center_y": .6}

	MDLabelLoader:
		id: txt
		halign: 'center'
		size_hint: (None, None)
		size: dp(250), dp(20)
		pos_hint: {"center_x": .5, "center_y": .4}

	MDRaisedButton:
		text: 'REMOVE' if txt.text else 'GET'
		pos_hint: {"center_x": .5, "center_y": .3}
		on_release:
			app.get(self)
"""


class MainApp(MDApp):
	def build(self):
		return Builder.load_string(KV)

	def get(self, widget):
		if widget.text == 'REMOVE':
			self.root.ids.txt.text = ''
			self.root.ids.img.source = ''
		else:
			self.root.ids.txt.text = 'Text'
			self.root.ids.img.source = 'your source'

if __name__ == "__main__":
	MainApp().run()
```
### Sample
light    dark

### CODE REVIEW

**Review by HeaTTheatR — COMMENTED** [review]
There is no `DataLoader` component in the material design specification.

File: kivymd/uix/loader.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+<MDImageLoader>:
+    source: ' '
+    canvas.before:
+        Color:
+            rgba: root.theme_cls.bg_darkest
+            a: root.fr_rec_opacity
+        RoundedRectangle:
+            pos: self.pos
+            size: self.size
+            radius: [(self.size[0]/2,self.size[1]/2) , (self.size[0]/2,self.size[1]/2),(self.size[0]/2,self.size[1]/2),(self.size[0]/2,self.size[1]/2) ] if root.circle==True else [(dp(20),dp(20)) , (dp(20),dp(20)) , (dp(20),dp(20)) , (dp(20),dp(20)) ]
```
Comment (HeaTTheatR) [inline kivymd/uix/loader.py:42]: Too long string

File: kivymd/uix/loader.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        Color:
+            rgba: root.theme_cls.bg_dark
+            a: root.bg_rec_opacity
+        RoundedRectangle:
+            pos: self.pos
+            size: self.size
+            radius: [(self.size[0]/2,self.size[1]/2) , (self.size[0]/2,self.size[1]/2),(self.size[0]/2,self.size[1]/2),(self.size[0]/2,self.size[1]/2) ] if root.circle==True else [(dp(20),dp(20)) , (dp(20),dp(20)) , (dp(20),dp(20)) , (dp(20),dp(20)) ]
```
Comment (HeaTTheatR) [inline kivymd/uix/loader.py:49]: Too long string

File: kivymd/uix/loader.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+"""
+)
+
+
+class MDLabelLoader(MDLabel):
+
+    bg_rec_opacity = NumericProperty(0)
+    fr_rec_opacity = NumericProperty(0)
```
Comment (HeaTTheatR) [inline kivymd/uix/loader.py:57]: Missing docstrings

File: kivymd/uix/loader.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        RoundedRectangle:
+            pos: self.pos
+            size: self.size
+            radius: [(self.size[0]/2,self.size[1]/2) , (self.size[0]/2,self.size[1]/2),(self.size[0]/2,self.size[1]/2),(self.size[0]/2,self.size[1]/2) ] if root.circle==True else [(dp(20),dp(20)) , (dp(20),dp(20)) , (dp(20),dp(20)) , (dp(20),dp(20)) ]
+"""
+)
+
+
+class MDLabelLoader(MDLabel):
+
```
Comment (HeaTTheatR) [inline kivymd/uix/loader.py:55]: Unnecessary empty string

File: kivymd/uix/loader.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    def on_text(self, *args):
+        if self._check_text(self.text):
+            self._stop_animate()
+        else:
+            self._start_animate()
+
+
+class MDImageLoader(ThemableBehavior, AsyncImage):
+
```
Comment (HeaTTheatR) [inline kivymd/uix/loader.py:109]: Unnecessary empty string

File: kivymd/uix/loader.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    bg_rec_opacity = NumericProperty(0)
+    fr_rec_opacity = NumericProperty(0)
+    circle = BooleanProperty(True)
```
Comment (HeaTTheatR) [inline kivymd/uix/loader.py:112]: Missing docstrings

**Comment by kulothunganug** [comment]
> There is no `DataLoader` component in the material design specification.

But i would be decorate kivymd and very useful
If you accept i will add docstrings

**Comment by HeaTTheatR** [comment]
@Kulothungan16 I think that the `AKivyMD` library is just created for these purposes - to supplement the `KivyMD` library with widgets that are not in the material design specification. Therefore, we can simply use `AKivyMD`.

**Comment by kulothunganug** [comment]
> @Kulothungan16 I think that the `AKivyMD` library is just created for these purposes - to supplement the `KivyMD` library with widgets that are not in the material design specification. Therefore, we can simply use `AKivyMD`.

That's correct 👍

**Pull request closed without merging**